netman/adapters/switches/dell.py

```diff
@@ -319,17 +319,52 @@ def read_interface(self, interface_name):
 
 def parse_vlan_list(result):
     vlans = []
+    vlan = None
     for line in result:
-        if regex.match('^(\d+)\s{1,6}(\S+).*', line):
-            number, name = regex
-            vlans.append(Vlan(number=int(number),
-                              name=name if int(number) > 1 else "default"))
-        elif regex.match('^(\d+)\s+.*', line):
-            number = regex[0]
-            vlans.append(Vlan(number=int(number)))
+        if regex.match('^(\d+)(.*)', line):
+            number, leftovers = regex
+            name = None
+            ports = None
+            if regex.match('^\s{1,6}(\S+)\s+([a-z0-9-,/]+)', leftovers):
+                name, ports = regex
+            elif regex.match('^\s{1,6}(\S+).*', leftovers):
+                name = regex[0]
+            elif regex.match('^\s*([a-z0-9-,/]+)', leftovers):
+                ports = regex[0]
+
+            vlan = Vlan(number=int(number),
+                        name=name if int(number) > 1 else "default")
+            vlans.append(vlan)
+            if ports:
+                vlan.interfaces.extend(parse_interface_list(ports))
+
+        elif regex.match('^\s+([a-z0-9-,/]+)', line):
+            vlan.interfaces.extend(parse_interface_list(regex[0]))
+
     return vlans
 
 
+def parse_interface_list(ports):
+    port_list = filter(None, ports.split(','))
+    interface_list = []
+    for port in port_list:
+        if regex.match('^ch([0-9]{1,2})-([0-9]{1,2})', port):
```

Comment:
You could simplify your regex with like \d+ instead of [0-9]{1,2}  it doesn't limit to 2 but do you really need to limit ? i think you can say anything that starts with chXXXXXXX with any number is valid

Same applis for all regexes